src/realm/obj.cpp

```diff
@@ -482,6 +419,90 @@ Mixed Obj::get_primary_key() const
         return Mixed{get_key()};
     }
 }
+template <class T>
+inline int Obj::cmp(const Obj& other, ColKey::Idx col_ndx) const
```

Comment:
Similar objections to combination of return and else as previously noted on the specialization for StringData

Reply:
Fixed